salt/personalised-covers/init.sls

```diff
@@ -0,0 +1,109 @@
+personalised-covers-repository:
+    builder.git_latest:
+        - name: git@github.com:elifesciences/personalised-covers.git
+        - identity: {{ pillar.elife.projects_builder.key or '' }}
+        - rev: {{ salt['elife.rev']() }}
+        - branch: {{ salt['elife.branch']() }}
+        - target: /srv/personalised-covers/
+        - force_fetch: True
+        - force_checkout: True
+        - force_reset: True
+        - fetch_pull_requests: True
+        - require:
+            - php-composer-1.0
+            - php-puli-latest
+
+    file.directory:
+        - name: /srv/personalised-covers
+        - user: {{ pillar.elife.deploy_user.username }}
+        - group: {{ pillar.elife.deploy_user.username }}
+        - recurse:
+            - user
+            - group
+        - require:
+            - builder: personalised-covers-repository
+
+# files and directories must be readable and writable by both elife and www-data
+# they are both in the www-data group, but the g+s flag makes sure that
+# new files and directories created inside have the www-data group
+# @source https://github.com/elifesciences/search-formula/blob/master/salt/search/init.sls#L42
```

Comment:
this works, but only in conunction with `umask(0002)` in all the bootstrap files of application code